src/pages/product-listing/product-listing.jsx

```diff
@@ -1,37 +1,71 @@
 import "./product-listing.css";
-import {useReducer} from "react";
 import { productsList } from "../../back-end/products-db";
-import {productListReducerFunc,getSortBy,getFilterByRating} from "../../functions/product-list-reducer.js"
+import {getSortBy,getFilterByRating,getFilterByCategory,getFilterByPrice} from "../../functions/filters-sorts.js";
+import { useProductListing } from "../../context/product-listing-context";
 const allProducts = productsList;
 const ProductListing = () =>
 {
-    const [state,dispatch] = useReducer(productListReducerFunc,{
-        rating:0,
-        sortBy:""
-    })
-    console.log(state.sortBy)
-    const sortList = getSortBy(allProducts,state.sortBy)
-    const finalProducts = getFilterByRating(sortList,state.rating)
-
+    const {state} = useProductListing()
+    const {dispatch} = useProductListing()
+    const categoryFilterProducts = getFilterByCategory(allProducts,state.categoryFilterBy.rings,state.categoryFilterBy.bangles,state.categoryFilterBy.necklace,
+        state.categoryFilterBy.earrings,state.categoryFilterBy.pendants)
+    const ratingFilteredProducts = getFilterByRating(categoryFilterProducts,state.rating)
+    const priceFilteredProducts = getFilterByPrice(ratingFilteredProducts,state.price)
+    const finalProducts = getSortBy(priceFilteredProducts,state.sortBy)
+    console.log(finalProducts)
     return(
         <div className="prod-page">
             <div className="aside">
                 <div className="filter">
                     <h3>Filters</h3>
-                    <p>Clear</p>
+                    <button onClick={() => dispatch({type:"CLEAR-FILTER"})}>Clear</button>
```

Comment:
button could use a little styling , it still seems to be the default button

Reply:
done